src/main/java/com/cloudbees/jenkins/GitHubRepositoryNameContributor.java

```diff
@@ -29,14 +33,40 @@
     /**
      * Looks at the definition of {@link AbstractProject} and list up the related github repositories,
      * then puts them into the collection.
+     *
+     * @deprecated Use {@link #parseAssociatedNames(Job, Collection)}
+     */
+    @Deprecated
+    public void parseAssociatedNames(AbstractProject<?,?> job, Collection<GitHubRepositoryName> result) {
+        parseAssociatedNames((Job) job, result);
+    }
+
+    /**
+     * Looks at the definition of {@link Job} and list up the related github repositories,
+     * then puts them into the collection.
      */
-    public abstract void parseAssociatedNames(AbstractProject<?,?> job, Collection<GitHubRepositoryName> result);
+    public /*abstract*/ void parseAssociatedNames(Job<?,?> job, Collection<GitHubRepositoryName> result) {
+        if (Util.isOverridden(GitHubRepositoryNameContributor.class, getClass(),
```

Comment:
would this not have been so much easier to just throw an exception in `parseAssociatedNames((AbstractProject<?,?> job, Collection<GitHubRepositoryName> result))` - no one could have called `super.` on it as it was abstract.
then this is purely just needing to call that method and let subclassed overrride this.